Mathlib/Topology/Sequences.lean

```diff
@@ -304,6 +304,40 @@ theorem CompactSpace.tendsto_subseq [CompactSpace X] (x : ℕ → X) :
 
 end FirstCountableTopology
 
+section Image
+
+variable {X Y : Type*} [TopologicalSpace X] [TopologicalSpace Y] {f : X → Y}
+
+/-- Sequential compactness of sets is preserved under sequentially continuous functions. -/
+theorem IsSeqCompact.image (f_cont : SeqContinuous f) {K : Set X} (K_cpt : IsSeqCompact K) :
+    IsSeqCompact (f '' K) := by
+  intro ys ys_in_fK
+  choose xs xs_in_K fxs_eq_ys using fun n ↦ (ys_in_fK n)
+  simp only [Set.mem_image, exists_exists_and_eq_and]
+  obtain ⟨a, a_in_K, phi, phi_mono, xs_phi_lim⟩ := K_cpt xs_in_K
+  refine ⟨a, a_in_K, phi, phi_mono, ?_⟩
+  exact Filter.Tendsto.congr (fun x ↦ fxs_eq_ys (phi x)) (f_cont xs_phi_lim)
+
+/-- The range of sequentially continuous function on a sequentially compact space is sequentially
+compact. -/
+theorem isSeqCompact_range [SeqCompactSpace X] (f_cont : SeqContinuous f) :
+    IsSeqCompact (Set.range f) := by
+  rw [← Set.image_univ]
+  exact IsSeqCompact.image f_cont ((seqCompactSpace_iff X).mp ‹SeqCompactSpace X›)
+
+/-- Sequential compactness of sets is preserved under continuous functions. -/
+theorem IsSeqCompact.image_of_continuous (f_cont : Continuous f) {K : Set X}
```

Comment:
I don't think that we need `_of_continuous` lemmas. It's easy enough to use `h.image f_cont.seqContinuous`.

Reply:
Let me comment briefly, since I think I was the one who suggested to @janemms the addition of the spellings with the assumption `Continuous f`.

I of course agree that instead of applying `IsSeqCompact.image_of_continuous` or `isSeqCompact_range_of_continuous`, it will be very easy to apply `IsSeqCompact.image` or `isSeqCompact_range` by transforming a continuity hypothesis to a sequential continuity hypothesis with dot notation.

The main reason why I thought that the versions `IsSeqCompact.image_of_continuous` or `isSeqCompact_range_of_continuous` would be useful is discoverability. For example Loogle would not find the main lemma `IsSeqCompact.image` with key words `IsSeqCompact` and `Continuous` (the users would need to generalize the search term to `SeqContinuous` by themselves.) Likewise, `exact?` would not find the lemmas, if the context only has continuity hypothesis (so a user would need to understand to add `have f_seqcont := f_cont.seqContinuous` before the search succeeds). Of course a sharp user will understand that only the general version should exist in Mathlib, but if generalizing the hypothesis does not even occur to the user, this will be easy to miss. It is different from cases where the generalized hypotheses are of type class form, since then `exact?` will do the appropriate generalization on behalf of the user.

I don't have strong opinions on which one is better, just wanted to present also the argument in favor of the choice to include the alternative versions as well.

Reply:
About `loogle`: I would search for `IsSeqCompact (_ '' _)`. About `exact?`: if `f` is given by a formula, then you need something like `continuity` anyway, so `exact?` won't work. Should we ask on Zulip for more opinions?

Reply:
E.g., we don't add `Continuous` versions of the lemmas assuming `Measurable` or `AEMeasurable`.

Reply:
To me either choice is fine. And I'm sure if your opinion after hearing the discoverability argument is clear, then that is almost surely the right thing to do!

(Regarding your last example, however, I find the existence of both `Measurable` and `AEMeasurable` versions of some lemmas is more analogous to the current `Continuous` and `SeqContinuous` duplication question. I think a user will not be surprised that measurability needs to be derived from continuity explicitly. But a user might not realize they should derive a.e.-measurability from measurability, so having both is good.)

Moreover, `SeqContinuous` and `IsSeqCompact` are probably going to be rather rare (much more so than `Measurable` and `AEMeasurable`). So I don't think the discoverability difficulties would affect very many users in any case. And I fully trust your judgement and experience about library design!

Reply:
After discussing this with @YaelDillies, I decided to remove the lemmas assuming continuity.